test/runtests.jl

```diff
@@ -87,13 +81,39 @@ end
     end
 end
 
+# Test separation between coordinates and conversion with mixed floating types.
+@testset "Position Angles" begin
+    c1 = ICRSCoords(0, 0)
+    c2 = ICRSCoords(deg2rad(1), 0)
+
+    # interface
+    @inferred position_angle(c1, c2)
+    @inferred position_angle(c1, c2 |> GalCoords)
+    @test position_angle(c1, c2) == position_angle(c1, c2 |> GalCoords)
```

Comment:
Maybe you want to keep inference test separate from the actual test, but note that you can combine everything in a single line:
```suggestion
    @test @inferred(position_angle(c1, c2)) == @inferred(position_angle(c1, c2 |> GalCoords))
```
However, this is probably a matter of taste, not super important.